pkg/dockerfile/generator_test.go

```diff
@@ -68,6 +68,116 @@ FROM python:3.8
+` + testInstallCog(gen.relativeTmpDir) + `
+WORKDIR /src
+CMD ["python", "-m", "cog.server.http"]
+COPY . /src`
+
+	require.Equal(t, expected, actual)
+}
+
+func TestBuildEnv(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "test")
+	require.NoError(t, err)
+
+	conf, err := config.FromYAML([]byte(`
+build:
+  gpu: false
+  env:
+    - XDG_CACHE_HOME=/src/custom_xdg_cache_home
+    - FOOBAR=foobar
+predict: cog_predict.py:Predictor
+`))
+	require.NoError(t, err)
+	require.NoError(t, conf.ValidateAndCompleteConfig())
+
+	gen, err := NewGenerator(conf, tmpDir)
+	require.NoError(t, err)
+	actual, err := gen.Generate()
+	require.NoError(t, err)
+
+	expected := `# syntax = docker/dockerfile:1.2
+FROM python:3.8
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu:/usr/local/nvidia/lib64:/usr/local/nvidia/bin
+ENV XDG_CACHE_HOME=/src/custom_xdg_cache_home
+ENV FOOBAR=foobar
+` + testInstallCog(gen.relativeTmpDir) + `
+WORKDIR /src
+CMD ["python", "-m", "cog.server.http"]
+COPY . /src`
+
+	require.Equal(t, expected, actual)
+}
+
+func TestBuildEnvEmpty(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "test")
+	require.NoError(t, err)
+
+	conf, err := config.FromYAML([]byte(`
+build:
+  gpu: false
+  env: []
+predict: cog_predict.py:Predictor
+`))
+	require.NoError(t, err)
+	require.NoError(t, conf.ValidateAndCompleteConfig())
+
+	gen, err := NewGenerator(conf, tmpDir)
+	require.NoError(t, err)
+	actual, err := gen.Generate()
+	require.NoError(t, err)
+
+	expected := `# syntax = docker/dockerfile:1.2
+FROM python:3.8
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu:/usr/local/nvidia/lib64:/usr/local/nvidia/bin
+ENV XDG_CACHE_HOME=/src/.cache
+` + testInstallCog(gen.relativeTmpDir) + `
+WORKDIR /src
+CMD ["python", "-m", "cog.server.http"]
+COPY . /src`
+
+	require.Equal(t, expected, actual)
+}
+
+func TestBuildEnvMultiple(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "test")
+	require.NoError(t, err)
+
+	conf, err := config.FromYAML([]byte(`
+build:
+  gpu: false
+  env:
+    - TRANSFORMERS_CACHE=/src/custom_huggingface_transformers_cache
+    - FOOBAR=foobar
+    - XDG_CACHE_HOME=/src/$FOOBAR
+    - RETICULATING=splines
+    - =invalid_doesnt_matter
+    - empty=
+predict: cog_predict.py:Predictor
+`))
+	require.NoError(t, err)
+	require.NoError(t, conf.ValidateAndCompleteConfig())
+
+	gen, err := NewGenerator(conf, tmpDir)
+	require.NoError(t, err)
+	actual, err := gen.Generate()
+	require.NoError(t, err)
+
+	expected := `# syntax = docker/dockerfile:1.2
+FROM python:3.8
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/x86_64-linux-gnu:/usr/local/nvidia/lib64:/usr/local/nvidia/bin
+ENV TRANSFORMERS_CACHE=/src/custom_huggingface_transformers_cache
+ENV FOOBAR=foobar
+ENV XDG_CACHE_HOME=/src/$FOOBAR
+ENV RETICULATING=splines
+ENV empty=
```

Comment:
I wonder if we could isolate testing the environment stuff with a separate function that is unit tested? If we wanted to change another part of the Dockerfile generation, we now have to update 3 additional tests.

Reply:
Good call, I'll give it a refactor.

Reply:
Refactored!